torch_xla/csrc/aten_cpu_fallback.cpp

```diff
@@ -0,0 +1,48 @@
+#include "torch_xla/csrc/aten_cpu_fallback.h"
+
+#include <tensorflow/compiler/xla/xla_client/debug_macros.h>
+#include <tensorflow/compiler/xla/xla_client/metrics.h>
+#include <tensorflow/compiler/xla/xla_client/tf_logging.h>
+#include <torch_xla/csrc/function_call_tracker.h>
+
+#include <unordered_map>
+
+namespace torch_xla {
+
+static std::unordered_map<std::string, ::xla::metrics::Counter*>
+    _cpu_fallback_counters;
+
+void xla_cpu_fallback(const c10::OperatorHandle& op, torch::jit::Stack* stack) {
+  XLA_FN_TRACK(3);
+  const auto name = c10::toString(op.operator_name());
+
+  // Manually applying the XLA_COUNTER macro.
+  // We need to do it ourselves and explicitly keep a mapping of counters
+  // because this boxed fallback kernel is used by multiple operators,
+  // and the macro stamps out a static Counter object with a fixed name
+  // at the code location that it was called.
+  if (_cpu_fallback_counters.find(name) == _cpu_fallback_counters.end()) {
+    _cpu_fallback_counters[name] = new ::xla::metrics::Counter(name);
+  }
+  _cpu_fallback_counters[name]->AddValue(1);
+
+  auto& args = op.schema().arguments();
+  auto arguments = torch::jit::last(stack, args.size());
+
+  // Log each tensor argument.
+  for (int64_t idx = 0; idx < arguments.size(); ++idx) {
+    const auto& ivalue = arguments[idx];
+    if (ivalue.isTensor()) {
+      TF_VLOG(3) << ivalue.toTensor().toString();
+    }
+  }
+
+  // Call the actual boxed CPU fallback.
```

Comment:
Will be nice to note here which device the tensor/ivalues are on here. (I assume it's still xla?

Reply:
Yup they should all be XLA (although technically if we had a meta function that allowed mixed device inputs, they could be mixed).

I could tack it onto `TF_VLOG(3) << ivalue.toTensor().toString();` if you think that would be useful. is `TF_VLOG(3)` the right macro to use for general purpose xla logging?